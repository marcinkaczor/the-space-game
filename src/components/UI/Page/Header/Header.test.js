/* global describe, test, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import PageHeader from './Header'

describe('PageHeader', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<PageHeader><div>Hello</div></PageHeader>)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with child and home and addition props', () => {
    const home = true
    const addition = 'Hello from addition'
    const Child = () => <div>Hello from child</div>

    const wrapper = mount(
      <PageHeader home={home} addition={addition}>
        <Child />
      </PageHeader>
    )
    expect(wrapper.find('h1').exists()).toEqual(home)
    expect(wrapper.find('h2').text()).toEqual(addition)
    expect(wrapper.find(Child).length).toEqual(1)
  })
})
