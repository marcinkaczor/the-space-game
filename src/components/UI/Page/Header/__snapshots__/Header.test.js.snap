// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader snapshot renders 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <h2
    className="MuiTypography-root MuiTypography-h2 MuiTypography-alignLeft"
  >
    <div>
      Hello
    </div>
  </h2>
</div>
`;
